File: ChatGPT/python/output/output_19a.py
# ChatGPT-generated solution will go here.
def main():
    with open("input.txt") as f:
        sections = f.read().strip().split("\n\n")

    towel_patterns = sections[0].split(", ")
    designs = sections[1].splitlines()

    towel_set = set(towel_patterns)
    memo = {}

    def can_form(design):
        if design in memo:
            return memo[design]
        if design == "":
            return True
        for pattern in towel_set:
            if design.startswith(pattern):
                if can_form(design[len(pattern):]):
                    memo[design] = True
                    return True
        memo[design] = False
        return False

    count = sum(1 for design in designs if can_form(design))
    print(count)

if __name__ == "__main__":
    main()

File: ChatGPT/Direct_Attempt/clojure/output/output_19a.clj
;; ChatGPT-generated solution will go here.
(ns output_19a
  (:require [clojure.string :as str]))

(defn parse-input [lines]
  (let [[patterns-line _ & designs] lines
        patterns (str/split patterns-line #",\s*")]
    {:patterns patterns
     :designs designs}))

(defn can-form? [patterns design]
  (let [n (count design)
        pattern-set (set patterns)
        dp (boolean-array (inc n))]
    (aset dp 0 true)
    (doseq [i (range 1 (inc n))]
      (doseq [j (range (max 0 (- i 10)) i)]
        (when (and (aget dp j)
                   (pattern-set (subs design j i)))
          (aset dp i true))))
    (aget dp n)))

(defn part [lines]
  (let [{:keys [patterns designs]} (parse-input lines)]
    (count (filter #(can-form? patterns %) designs))))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
